pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -540,10 +540,6 @@ func validateResultsFromMatrixedPipelineTasksNotConsumed(tasks []PipelineTask, f
 // error is returned when the array indexing reference is out of bound of the array param
 // e.g. if a param reference of $(params.array-param[2]) and the array param is of length 2.
 func (ps *PipelineSpec) ValidateParamArrayIndex(ctx context.Context, params Params) error {
-	if !config.CheckAlphaOrBetaAPIFields(ctx) {
```

Comment:
not in scope for this PR, but this function looks redundant with this one: https://github.com/tektoncd/pipeline/blob/main/pkg/apis/version/version_validation.go#L30

It would be good to remove `CheckAlphaOrBetaAPIFields`

Reply:
`CheckAlphaOrBetaAPIFields` is equal to 
```
ValidateEnabledAPIFields(ctx,"","alpha") || ValidateEnabledAPIFields(ctx,"","alpha")
```
So I think using one function to check alpha or beta is better?  And ValidateEnabledAPIFields is mainly used for validation webhook.

Reply:
I agree that one function would be better! `CheckAlphaOrBetaAPIFields` is equivalent to `ValidateEnabledAPIFields(ctx, "", "beta")`, which validates that the flag is set to beta stability or lower.